### PR TITLE
fix: cpp-lint errors, and other changes

### PR DESCRIPTION
**Pull request by AbhishekAjoy**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTION.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTION.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [ ] Sort by alphabetical order
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: sorting/pancake_sort.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2020 @author Abhishek Ajoy Menon
+ *
+ * @file
+ *
+ * A generic implementation of pancake_sort.
+ */
+#include<iostream>
+// Sorting an array of integers in ascending order
```
Comment (Panquesito7) [inline sorting/pancake_sort.cpp:9]: Please use

```c++
/**
 * Something here
 */
```
except inside functions.

File: sorting/pancake_sort.cpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+/**
+ * The algorithm finds the largest number in the array
+ * and moves it to the end of the array.
+ * This moving of array to the end is done by reversing the array
+ * from the start to the location of the largest number (which 
+ * brings the largest number to the beginning) and then reversing 
+ * the whole array. The reversing process is analogous to flipping
+ * pancakes, hence the name.
+*/
+// Reverses arr[0..i]
```
Comment (Panquesito7) [inline sorting/pancake_sort.cpp:19]: Same here as above and in other parts of the code..

File: .vscode/launch.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{
```
Comment (kvedala) [inline .vscode/launch.json:1]: please delete this file.

File: .vscode/tasks.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (kvedala) [inline .vscode/tasks.json:1]: please delete this file

File: sorting/pancake_sort.cpp [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    int temp, start = 0;
+    while (start < i) {
+        temp = arr[start];
+        arr[start] = arr[i];
+        arr[i] = temp;
+        start++;
+        i--;
+    }
+}
+// Returns index of the  maximum element in   arr[0..n-1]
```
Comment (kvedala) [inline sorting/pancake_sort.cpp:30]: please follow the doxygen documentation protocol as explained [here](https://www.doxygen.nl/manual/docblocks.html). You can use #807 as example.

**Comment by AbhishekAjoy** [comment]
what does sort by alphabetical order checkbox mean on the description of change.

File: .vscode/tasks.json (hunk 4 above)
Comment (kvedala) [reply .vscode/tasks.json:1]: ☝️

File: sorting/pancake_sort.cpp [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+ * The algorithm finds the largest number in the array
+ * and moves it to the end of the array.
+ * This moving of array to the end is done by reversing the array
+ * from the start to the location of the largest number (which 
+ * brings the largest number to the beginning) and then reversing 
+ * the whole array. The reversing process is analogous to flipping
+ * pancakes, hence the name.
+*/
+/**
+ *  flip(): Reverses arr[0..i]
```
Comment (kvedala) [inline sorting/pancake_sort.cpp:20]:
```suggestion
 * Reverses elements of an input array
 * \param [in,ou] arr array to reverse
 * \param[in] i length of array
```

File: sorting/pancake_sort.cpp [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+ */
+void flip(int arr[], int i) {
```
Comment (kvedala) [inline sorting/pancake_sort.cpp:22]:
```suggestion
template<typename T>
void flip(T *arr, int i) {
```

File: sorting/pancake_sort.cpp [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    int temp, start = 0;
```
Comment (kvedala) [inline sorting/pancake_sort.cpp:23]:
```suggestion
    int start = 0;
    T temp;
```

File: sorting/pancake_sort.cpp [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+    while (start < i) {
+        temp = arr[start];
```
Comment (kvedala) [inline sorting/pancake_sort.cpp:25]: can't you use `std::swap(arr[start], arr[i])` and save 4 lines?

File: sorting/pancake_sort.cpp [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+        arr[start] = arr[i];
+        arr[i] = temp;
+        start++;
+        i--;
+    }
+}
+/**
+ * findMax(): Returns index of the  maximum element in arr[0..n-1]
+ */
+int findMax(int arr[], int n) {
```
Comment (kvedala) [inline sorting/pancake_sort.cpp:35]: see changes suggested above for documentation of function and function arguments

File: sorting/pancake_sort.cpp [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+    int mi, i;
+    for (mi = 0, i = 0; i < n; ++i)
+    if (arr[i] > arr[mi])
+    mi = i;
+    return mi;
+}
+/** 
+ * pancakeSort(): The main function that sorts given array using flip operations
+ */
+void pancakeSort(int *arr, int n) {
```
Comment (kvedala) [inline sorting/pancake_sort.cpp:45]: see changes suggested above for documentation of function and function arguments

File: sorting/pancake_sort.cpp [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+}
+/**
+ * main():
+ * Reads the number of elements in the array 
+ * and then reads the elements. The array is sorted by calling the 
+ * pancakeSort function. The sorted array is then displayed.
+*/
+int main() {
+    int arr[20];
+    int n;
```
Comment (kvedala) [inline sorting/pancake_sort.cpp:69]: please add test-cases as done in: https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/sorting/heap_sort.cpp

**Comment by kvedala** [comment]
@AbhishekAjoy any update on this? Thanks

**Pull request closed without merging**